### ISSUE TITLE
dev container is unable to build docs in bind mounted directory

### ISSUE THREAD

**Issue by gsnider2195** [nautobot/nautobot#4164]
### Environment
* Nautobot version (Docker tag too if applicable): 2.0.0b2
* Python version: 3.10.12 (3.8.17 in container)
* Database platform, version: N/A
* Middleware(s):

### Steps to Reproduce
1. Build docker container with latest version of `next` (after #4130)
2. Try to run `inv build-and-check-docs`

### Expected Behavior
Docs are built

### Observed Behavior
Permission denied. mkdocs tries to build the docs in `$PWD/project-static/docs` which is owned by the user on the host system (uid 1000 for me) but is running as uid 999 in the container

My quick workaround was to run `build_nautobot_docs` and `build_example_plugin_docs` as the current user:

```diff
diff --git a/tasks.py b/tasks.py
index 9dc59ee2d..e60fb0eaa 100644
--- a/tasks.py
+++ b/tasks.py
@@ -536,7 +536,11 @@ def build_and_check_docs(context):
 def build_nautobot_docs(context):
     "Build Nautobot docs."
     command = "mkdocs build --no-directory-urls --strict"
-    run_command(context, command)
+    if is_truthy(context.nautobot.local):
+        run_command(context, command)
+    else:
+        docker_command = f"run --user {os.getuid()}:{os.getgid()} --entrypoint '{command}' nautobot"
+        docker_compose(context, docker_command, pty=True)
 
 
 def build_example_plugin_docs(context):
@@ -547,7 +551,7 @@ def build_example_plugin_docs(context):
         print_command(local_command)
         context.run(local_command, pty=True)
     else:
-        docker_command = f"run --workdir='/source/examples/example_plugin' --entrypoint '{command}' nautobot"
+        docker_command = f"run --user {os.getuid()}:{os.getgid()} --workdir='/source/examples/example_plugin' --entrypoint '{command}' nautobot"
         docker_compose(context, docker_command, pty=True)
```

I'm still getting permission denied when unittest tries to create `.coverage` so we may need to run unittest as the local user also.

**Comment by gsnider2195** [comment]
My preferred alternate option is to continue to run as root in the `dev` stage. Thoughts @glennmatthews?

**Comment by glennmatthews** [comment]
I'm not necessarily opposed to that option. I was kinda worried that we might hit issues like this with #4130 given the difference in how Docker Desktop works on Mac versus other platforms. Whatever the cleanest/quickest way to fix this turns out to be is fine with me.

**Comment by gsnider2195** [comment]
I'd like to propose that we stick with root in the dev stage for now. Even with the workaround proposed above, after switching to the `develop` branch, the docs get created by root so when switching back to `next` I have to `rm -rf examples/example_plugin/example_plugin/static/example_plugin/docs` and `rm -rf nautobot/project-static/docs` to get back to a working state.

**Comment by gsnider2195** [comment]
Resolved by #4213
